In [1]:
#Task 1
import pandas as pd
print(pd.__version__)

2.3.3


In [2]:
#Task 2
customers_df = pd.read_csv("customers.csv")
customers_df

,age,income,visits_per_month
0,30.483571,33893.885591,9.295377
1,35.615149,33126.773002,7.531726
2,35.896064,41139.477833,7.061051
3,30.712800,31292.658458,7.068540
4,29.209811,19693.758043,4.550164
...,...,...,...
295,67.926951,103758.922780,2.640708
296,59.134584,111630.553709,3.820482
297,58.550918,111000.120344,4.169296
298,64.675113,104730.648314,2.832882


In [5]:
# Task 3
print(customers_df.head(6))

         age        income  visits_per_month
0  30.483571  33893.885591          9.295377
1  35.615149  33126.773002          7.531726
2  35.896064  41139.477833          7.061051
3  30.712800  31292.658458          7.068540
4  29.209811  19693.758043          4.550164
5  25.188562  26897.351037          8.628495


In [6]:
print(customers_df.tail(4))

           age         income  visits_per_month
296  59.134584  111630.553709          3.820482
297  58.550918  111000.120344          4.169296
298  64.675113  104730.648314          2.832882
299  56.026996  113097.634498          2.183064


In [9]:
print(customers_df.sample(5))

           age        income  visits_per_month
109  44.749976  72489.437805          5.942518
273  62.458363  87240.673248          4.409347
141  45.829426  72744.511395          7.052947
174  36.024257  71847.184023          3.855111
288  50.710479  89184.476601          3.285865


Sample DOES return different rows each time because the results are random

In [11]:
#Task 4
cars_df = pd.read_csv("cars.csv")
cars_df

,mpg,cylinders,displacement,horsepower,weight,acceleration,model year,origin,car name
0,18.0,8,307.0,130,3504,12.0,70,1,chevrolet chevelle malibu
1,15.0,8,350.0,165,3693,11.5,70,1,buick skylark 320
2,18.0,8,318.0,150,3436,11.0,70,1,plymouth satellite
3,16.0,8,304.0,150,3433,12.0,70,1,amc rebel sst
4,17.0,8,302.0,140,3449,10.5,70,1,ford torino
...,...,...,...,...,...,...,...,...,...
393,27.0,4,140.0,86,2790,15.6,82,1,ford mustang gl
394,44.0,4,97.0,52,2130,24.6,82,2,vw pickup
395,32.0,4,135.0,84,2295,11.6,82,1,dodge rampage
396,28.0,4,120.0,79,2625,18.6,82,1,ford ranger


Rows?  -- 398
Columns? --9
Column Names? -- mpg	cylinders	displacement	horsepower	weight	acceleration	model year	origin	car name

In [12]:
# Task 5
cars_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    398 non-null    object 
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model year    398 non-null    int64  
 7   origin        398 non-null    int64  
 8   car name      398 non-null    object 
dtypes: float64(3), int64(4), object(2)
memory usage: 28.1+ KB


Task 5
Whole Numbers: cylinders, weight, model year, origin
Decimal Numbers: mpg, displacement, acceleration
Object: horsepower, car name

In [14]:
#Task 6
titanic_df = pd.read_csv("titanic_survival.csv")
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


In [ ]:
Rows = 891  columns = 15
most missing values: deck
You know it is missing because of the number of non-null values in the deck column is 203, which is lower than the number of rows...by a lot

In [15]:
#Task 7
cars_df.describe()

,mpg,cylinders,displacement,weight,acceleration,model year,origin
count,398.000000,398.000000,398.000000,398.000000,398.000000,398.000000,398.000000
mean,23.514573,5.454774,193.425879,2970.424623,15.568090,76.010050,1.572864
std,7.815984,1.701004,104.269838,846.841774,2.757689,3.697627,0.802055
min,9.000000,3.000000,68.000000,1613.000000,8.000000,70.000000,1.000000
25%,17.500000,4.000000,104.250000,2223.750000,13.825000,73.000000,1.000000
50%,23.000000,4.000000,148.500000,2803.500000,15.500000,76.000000,1.000000
75%,29.000000,8.000000,262.000000,3608.000000,17.175000,79.000000,2.000000
max,46.600000,8.000000,455.000000,5140.000000,24.800000,82.000000,3.000000


Mean of mpg = 23.514573
Max horsepower = ??
Median weight of car: 2803.5

In [23]:
#Task 8
titanic_df['fare'].describe()

count    891.000000
mean      32.204208
std       49.693429
min        0.000000
25%        7.910400
50%       14.454200
75%       31.000000
max      512.329200
Name: fare, dtype: float64

The fare data has outliers. You can tell because the mean and median are pretty different.

In [ ]:
#Task 9
import pandas as pd
data = {
    'product': ['pencil', 'pen', 'eraser', 'notebook'],
    ''
}